crypto/signatures_test.go

```diff
@@ -276,3 +276,59 @@ func TestIntegrationSigKeyGeneration(t *testing.T) {
 		t.Error("corruption failed")
 	}
 }
+
+// TestSignVerifyObject tests the SignObject and VerifyObject functions, which
+// are inverses of each other.
+func TestSignVerifyObject(t *testing.T) {
+	sk, pk, err := GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create signed object
+	signedObj := SignObject("foo", sk)
+
+	// verify signed object
+	var read string
+	err = VerifyObject(signedObj, &read, pk)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if read != "foo" {
+		t.Fatal("encode/decode mismatch: expected 'foo', got", []byte(read))
+	}
+
+	// corrupt signature
+	_, err = rand.Read(signedObj[11:])
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = VerifyObject(signedObj, &read, pk)
+	if err != errInvalidSignature {
+		t.Fatal("expected errInvalidSignature, got", err)
+	}
+
+	// not enough object data
+	err = VerifyObject(signedObj[:0], &read, pk)
+	if err == nil || err.Error() != "could not decode type []uint8: EOF" {
+		t.Fatal("expected decode error, got", err)
+	}
+	// not enough signature data
+	err = VerifyObject(signedObj[:19], &read, pk)
+	if err == nil || err.Error() != "could not decode type crypto.Signature: EOF" {
+		t.Fatal("expected decode error, got", err)
+	}
```

Comment:
I would feel better if we wrapped this whole function in a `for` loop, especially given that the sign and verify functions have failed nondeterministically for previous implementations (to the tune of 1-in-200 times or so).

```
var j int
if testing.Short {
    j = 3
} else {
    j = 200
}
for i := 0; i < j; i++ {
    // whole test
}
```